Source/Serilog.Exceptions/Core/ExceptionPropertiesBag.cs

```diff
@@ -0,0 +1,46 @@
+namespace Serilog.Exceptions.Core
+{
+    using System;
+    using System.Collections.Generic;
+
+    internal class ExceptionPropertiesBag : IExceptionPropertiesBag
+    {
+        private readonly Type exceptionType;
+        private readonly IExceptionPropertyFilter filter;
+        private readonly Dictionary<string, object> properties = new Dictionary<string, object>();
+        private bool resultsCollected = false;
+
+        public ExceptionPropertiesBag(Type exceptionType, IExceptionPropertyFilter filter = null)
+        {
+            this.exceptionType = exceptionType;
+            this.filter = filter;
+        }
+
+        public IReadOnlyDictionary<string, object> ResultDictionary
+        {
+            get
+            {
+                this.resultsCollected = true;
+                return this.properties;
+            }
+        }
+
+        public void AddProperty(string key, object value)
```

Comment:
Lets add a check for null here and throw `ArgumentNullException` since this is a public facing API.

Reply:
Done